File: android/app/src/main/java/com/quietly/app/data/model/Book.kt
package com.quietly.app.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Book(
    val id: String,
    val title: String,
    val author: String,
    val isbn: String? = null,
    @SerialName("cover_url")
    val coverUrl: String? = null,
    @SerialName("page_count")
    val pageCount: Int? = null,
    val description: String? = null,
    @SerialName("published_date")
    val publishedDate: String? = null,
    val publisher: String? = null,
    @SerialName("open_library_key")
    val openLibraryKey: String? = null,
    @SerialName("created_at")
    val createdAt: String? = null
)

File: ios/QuietlyApp/QuietlyApp/Models/Book.swift
import Foundation
import SwiftData

@Model
final class Book {
    @Attribute(.unique) var id: UUID
    var isbn: String?
    var title: String
    var author: String?
    var coverUrl: String?
    var publisher: String?
    var publishedDate: String?
    var bookDescription: String?
    var pageCount: Int?
    var manualEntry: Bool
    var createdAt: Date

    // Relationships
    @Relationship(deleteRule: .cascade, inverse: \UserBook.book)
    var userBooks: [UserBook] = []

    @Relationship(deleteRule: .cascade, inverse: \Note.book)
    var notes: [Note] = []

    @Relationship(deleteRule: .cascade, inverse: \ReadingSession.book)
    var sessions: [ReadingSession] = []

    init(
        id: UUID = UUID(),
        isbn: String? = nil,
        title: String,
        author: String? = nil,
        coverUrl: String? = nil,
        publisher: String? = nil,
        publishedDate: String? = nil,
        bookDescription: String? = nil,
        pageCount: Int? = nil,
        manualEntry: Bool = false,
        createdAt: Date = Date()
    ) {
        self.id = id
        self.isbn = isbn
        self.title = title
        self.author = author
        self.coverUrl = coverUrl
        self.publisher = publisher
        self.publishedDate = publishedDate
        self.bookDescription = bookDescription
        self.pageCount = pageCount
        self.manualEntry = manualEntry
        self.createdAt = createdAt
    }
}
